File: cmd/check.go
package cmd

import (
	"fmt"
	"zoi/src"

	"github.com/spf13/cobra"
)

var checkCmd = &cobra.Command{
	Use:   "check",
	Short: "Verify system health and check for tool requirements",
	Run: func(cmd *cobra.Command, args []string) {
		toolsToCheck := []string{
			"git",
		}

		yellow := src.Yellow()
		allChecksPassed := true

		src.PrintHighlight("--- Checking for Essential Tools ---")

		for _, tool := range toolsToCheck {
			fmt.Printf("Checking for %s... ", yellow.Sprint(tool))
			if src.CheckCommand(tool) {
				src.PrintSuccess("OK")
			} else {
				src.PrintError("MISSING")
				allChecksPassed = false
			}
		}

		fmt.Println()

		if allChecksPassed {
			src.PrintSuccess("All essential tools checked are installed.")
		} else {
			src.PrintError("Some required tools are missing. Please install them to ensure full functionality.")
		}
	},
}

func init() {
	rootCmd.AddCommand(checkCmd)
}

File: src/cmd/check.rs
use crate::utils;
use anyhow::{Result, anyhow};
use colored::*;
use std::io::{Write, stdout};

pub fn run() -> Result<()> {
    println!("{}", "--- Checking for Essential Tools ---".yellow().bold());

    let essential_tools = ["git"];
    let mut all_essential_found = true;

    for tool in essential_tools {
        print!("Checking for {}... ", tool.cyan());
        let _ = stdout().flush();

        if utils::command_exists(tool) {
            println!("{}", "OK".green());
        } else {
            println!("{}", "MISSING".red());
            all_essential_found = false;
        }
    }

    if !all_essential_found {
        return Err(anyhow!(
            "One or more essential tools are missing. Please install them."
        ));
    }

    println!(
        "
{}",
        "--- Checking for Recommended Tools ---".yellow().bold()
    );
    let recommended_tools = ["less", "gpg"];
    for tool in recommended_tools {
        print!("Checking for {}... ", tool.cyan());
        let _ = stdout().flush();

        if utils::command_exists(tool) {
            println!("{}", "OK".green());
        } else {
            println!("{}", "Not Found".yellow());
        }
    }

    println!();

    println!("{}", "All essential tools are installed.".green());
    Ok(())
}
